server/queue/queue-rabbitmq/src/main/java/org/apache/james/queue/rabbitmq/Dequeuer.java

```diff
@@ -0,0 +1,182 @@
+
+import javax.mail.MessagingException;
+import javax.mail.internet.AddressException;
+import javax.mail.internet.MimeMessage;
+
+import org.apache.commons.lang3.tuple.Pair;
+import org.apache.james.blob.api.BlobId;
+import org.apache.james.blob.api.Store;
+import org.apache.james.blob.mail.MimeMessagePartsId;
+import org.apache.james.core.MailAddress;
+import org.apache.james.metrics.api.Metric;
+import org.apache.james.metrics.api.MetricFactory;
+import org.apache.james.queue.api.MailQueue;
+import org.apache.james.server.core.MailImpl;
+import org.apache.james.util.SerializationUtil;
+import org.apache.mailet.Mail;
+import org.apache.mailet.PerRecipientHeaders;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.github.fge.lambdas.Throwing;
+import com.github.steveash.guavate.Guavate;
+import com.nurkiewicz.asyncretry.AsyncRetryExecutor;
+import com.rabbitmq.client.GetResponse;
+
+class Dequeuer {
+    private static class NoMailYetException extends RuntimeException {
+    }
+
+    private static class RabbitMQMailQueueItem implements MailQueue.MailQueueItem {
+        private final RabbitClient rabbitClient;
+        private final long deliveryTag;
+        private final Mail mail;
+
+        private RabbitMQMailQueueItem(RabbitClient rabbitClient, long deliveryTag, Mail mail) {
+            this.rabbitClient = rabbitClient;
+            this.deliveryTag = deliveryTag;
+            this.mail = mail;
+        }
+
+        @Override
+        public Mail getMail() {
+            return mail;
+        }
+
+        @Override
+        public void done(boolean success) throws MailQueue.MailQueueException {
+            try {
+                rabbitClient.ack(deliveryTag);
+            } catch (IOException e) {
+                throw new MailQueue.MailQueueException("Failed to ACK " + mail.getName() + " with delivery tag " + deliveryTag, e);
+            }
+        }
+    }
+
+    private static final int TEN_MS = 10;
+
+    private final MailQueueName name;
+    private final RabbitClient rabbitClient;
+    private final Store<MimeMessage, MimeMessagePartsId> mimeMessageStore;
+    private final BlobId.Factory blobIdFactory;
+    private final ObjectMapper objectMapper;
+    private final Metric dequeueMetric;
+
+    Dequeuer(MailQueueName name, RabbitClient rabbitClient, Store<MimeMessage, MimeMessagePartsId> mimeMessageStore, BlobId.Factory blobIdFactory, ObjectMapper objectMapper, MetricFactory metricFactory) {
+        this.name = name;
+        this.rabbitClient = rabbitClient;
+        this.mimeMessageStore = mimeMessageStore;
+        this.blobIdFactory = blobIdFactory;
+        this.objectMapper = objectMapper;
+        this.dequeueMetric = metricFactory.generate(DEQUEUED_METRIC_NAME_PREFIX + name.asString());
+    }
+
+    MailQueue.MailQueueItem deQueue() throws MailQueue.MailQueueException {
+        GetResponse getResponse = pollChannel();
+        MailDTO mailDTO = toDTO(getResponse);
```

Comment:
These two steps don't make sense here, we don't need MailDTO at all in this method. what about a method that would convert GetResponse to Mail directly ?